src/marketplace/create_listings.ts

```diff
@@ -1,196 +1,277 @@
-      });
+    if (!isValidPrivateKey(privateKey) && privateKey) {
+      console.log('Please input a valid EVM Private key');
+      process.exit();
     }
-  
-    if (!price) {
-      price = await input({
-        message: "Enter the price per token, in the currency specified",
-      });
-    }
-
-    const chainConfig = findSupportedNetwork(network);
+  }
 
-    if (chainConfig === undefined) {
-        program.error("Unsupported network, please select a valid network")
-        return
-    }
+  if (!collectionAddress) {
+    collectionAddress = await input({
+      message: 'Enter the address of the collection contract',
+    });
+  }
 
-    let nodeUrl = chainConfig.rpcUrl
+  if (tokenId === null || tokenId === undefined) {
+    tokenId = await number({
+      message: 'Enter the ID for the token to be listed',
+    });
+  }
 
-    if (options.projectAccessKey) {
-        nodeUrl += "/" + options.projectAccessKey
-    }
-    
-    if (options.verbose) {
-      console.log(`Using node URL: ${nodeUrl}`)
-    }
-  
-    const provider = new ethers.JsonRpcProvider(nodeUrl);
-    const wallet = new ethers.Wallet(privateKey, provider);
-  
-    console.log(`Using EOA Wallet: ${wallet.address}`);
-    console.log("Collection Address:", collectionAddress);
-    console.log("Currency token address:", currency);
-  
-    const collectionContract = new ethers.Contract(
-      collectionAddress,
-      ERC1155_ABI,
-      wallet
+  if (!network) {
+    console.log(
+      'Please provide the Network for your project as a Sequence chain handle'
+    );
+    console.log(
+      'Possible networks can be found at https://docs.sequence.xyz/solutions/technical-references/chain-support'
     );
 
-    const balance = await collectionContract.balanceOf(wallet.address, tokenId);
-  
-    let isApprovedForAll = false;
-  
+    network = await input({
+      message: 'Enter the network to be used (mainnet, polygon, etc.)',
+    });
+  }
+
+  if (!currency) {
+    currency = await input({
+      message:
+        'Enter either a well known currency code (USDC, MATIC, etc.) or a token address',
+    });
+  }
+
+  if (!price) {
+    price = await input({
+      message: 'Enter the price per token, in the currency specified',
+    });
+  }
+
+  const chainConfig = findSupportedNetwork(network);
+  if (chainConfig === undefined) {
+    program.error('Unsupported network, please select a valid network');
+    return;
+  }
+
+  let nodeUrl = chainConfig.rpcUrl;
+
+  if (options.projectAccessKey) {
+    nodeUrl += '/' + options.projectAccessKey;
+  }
+
+  if (options.verbose) {
+    console.log(`Using node URL: ${nodeUrl}`);
+  }
+
+  const provider = new ethers.JsonRpcProvider(nodeUrl);
+  const wallet = new ethers.Wallet(privateKey, provider);
+
+  // Set up the wallet client
+  const walletClient = createWalletClient({
+    account: privateKeyToAccount(privateKey),
+    chain: {
+      id: chainConfig.chainId,
+      name: chainConfig.name,
```

Comment:
```suggestion
      name: chainConfig.title,
```
viem uses the name to mean a human-readable name, while sequence uses name to mean 'sequence service name'